packages/payment/src/services/payment-module.ts

```diff
@@ -1,29 +1,283 @@
+  SetPaymentSessionsDTO,
+  UpdatePaymentCollectionDTO,
+  UpdatePaymentDTO,
 } from "@medusajs/types"
+import {
+  InjectManager,
+  InjectTransactionManager,
+  MedusaContext,
+} from "@medusajs/utils"
 
 import { Payment } from "@models"
+import * as services from "@services"
 
 import { joinerConfig } from "../joiner-config"
 
 type InjectedDependencies = {
   baseRepository: DAL.RepositoryService
+  paymentCollectionService: services.PaymentCollection
 }
 
-// TODO: implement IPaymentModule
-export default class PaymentModule<TPayment extends Payment = Payment> {
+export default class PaymentModule<TPayment extends Payment = Payment>
+  implements IPaymentModuleService
+{
   protected baseRepository_: DAL.RepositoryService
+  protected paymentCollectionService_: services.PaymentCollection
 
   constructor(
-    { baseRepository }: InjectedDependencies,
+    { baseRepository, paymentCollectionService }: InjectedDependencies,
     protected readonly moduleDeclaration: InternalModuleDeclaration
   ) {
     this.baseRepository_ = baseRepository
+
+    this.paymentCollectionService_ = paymentCollectionService
   }
 
   __joinerConfig(): ModuleJoinerConfig {
     return joinerConfig
   }
+
+  createPaymentCollection(
+    data: CreatePaymentCollectionDTO,
+    sharedContext?: Context
+  ): Promise<PaymentCollectionDTO>
+
+  createPaymentCollection(
+    data: CreatePaymentCollectionDTO[],
+    sharedContext?: Context
+  ): Promise<PaymentCollectionDTO[]>
+
+  @InjectTransactionManager("baseRepository_")
+  async createPaymentCollection(
+    data: CreatePaymentCollectionDTO | CreatePaymentCollectionDTO[],
+    @MedusaContext() sharedContext?: Context
+  ): Promise<PaymentCollectionDTO | PaymentCollectionDTO[]> {
+    const input = Array.isArray(data) ? data : [data]
+
+    const collections = await this.paymentCollectionService_.create(
+      input,
+      sharedContext
+    )
+
+    return await this.baseRepository_.serialize<PaymentCollectionDTO[]>(
+      Array.isArray(data) ? collections : collections[0],
+      {
+        populate: true,
+      }
+    )
+  }
+
+  updatePaymentCollection(
+    data: UpdatePaymentCollectionDTO[],
+    sharedContext?: Context
+  ): Promise<PaymentCollectionDTO[]>
+  updatePaymentCollection(
+    data: UpdatePaymentCollectionDTO,
+    sharedContext?: Context
+  ): Promise<PaymentCollectionDTO>
+
+  @InjectTransactionManager("baseRepository_")
+  async updatePaymentCollection(
+    data: UpdatePaymentCollectionDTO | UpdatePaymentCollectionDTO[],
+    sharedContext?: Context
+  ): Promise<PaymentCollectionDTO | PaymentCollectionDTO[]> {
+    const input = Array.isArray(data) ? data : [data]
+    const result = await this.paymentCollectionService_.update(
+      input,
+      sharedContext
+    )
+
+    return await this.baseRepository_.serialize<PaymentCollectionDTO[]>(
+      Array.isArray(data) ? result : result[0],
+      {
+        populate: true,
+      }
+    )
+  }
+
+  deletePaymentCollection(
```

Comment:
suggestion: Add selector-based deletion

Reply:
Could you elaborate please, doesn't `AbstractService_` allow for deletion by primary key ATM?

Reply:
I will add the overload and implementation soon, but yes for now you are right cc @olivermrbl